edx_proctoring/api.py

```diff
@@ -1054,9 +1063,22 @@ def create_exam_attempt(exam_id, user_id, taking_as_proctored=False):
         )
         raise StudentExamAttemptOnPastDueProctoredExam(err_msg)
 
+    should_create_verified_name = False
+
     if taking_as_proctored:
-        external_id, force_status = _register_proctored_exam_attempt(
-            user_id, exam_id, exam, attempt_code, review_policy
+        verified_name = None
+        if is_verified_name_enabled:
+            # If the user has a verified name, use it instead of their profile name for
+            # the backend's ID verification. If they don't have one, create one with the
+            # exam attempt.
+            user = USER_MODEL.objects.get(id=user_id)
+            verified_name_obj = get_verified_name(user)
+            if not verified_name_obj or not verified_name_obj.is_verified:
+                should_create_verified_name = True
+            if verified_name_obj:
+                verified_name = verified_name_obj.verified_name
```

Comment:
in the case where verified_name_obj exists but it is_verified = false, we do should_create_verified name which seems right to make the exam attempt, but then we use verified_name_obj.verified_name which is not verified

that doesn't seem right

Reply:
This is so that, if the user has a "pending" verified name (IE, they're in the middle of ID verification), they can still use it for a proctored exam attempt. (thank you for bringing this up, I should leave a comment so that the use case is clearer)

Reply:
there is a test for this case but it's not checking how _register_protctored is called as far as I can tell